tests/lax_numpy_test.py

```diff
@@ -1473,6 +1473,20 @@ def testDiagFlat(self, shape, dtype, k):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=True)
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}".format(
+          jtu.format_shape_dtype_string(shape, dtype)),
+       "dtype": dtype, "shape": shape}
```

Comment:
We should test inputs of differing shapes to get better test coverage. You could use `a1_shape` and `a2_shape`.